src/decaylanguage/dec/dec.py

```diff
@@ -281,18 +294,80 @@ def load_grammar(
         for parser, lexer and options.
         """
 
+        warnings.warn(
+            "This method is deprecated, please add unknown decay models by passing them to "
+            "`load_additional_decay_models` instead before parsing the decayfile.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+
         if filename is None:
             filename = "decfile.lark"
-            with data.basepath.joinpath(filename).open() as f1:
-                self._grammar = f1.read()
+
+        self._load_grammar(filename, parser, lexer, **options)
+
+    def load_additional_decay_models(self, *models: str) -> None:
```

Comment:
This method requires a docstring, even more because it is a public method.

The code seems to imply that  `models` is only taken into account if `self._additional_decay_models ` is None. That is most probably the default usage. But imagine somebody does provide additional models in the `__init__` method already; then this call will override, right? One should probably expand the test below to confirm and ensure the code does what it is expected in such a case. In fact, one should give examples in https://github.com/scikit-hep/decaylanguage/blob/master/README.md#advanced-usage. Let me know if you have limited time and prefer that we do this as a follow-up, which is OK.

Reply:
Ah, thank you for catching this, I'll add a docstring. However, I have to correct the second part (if I have understood you correctly): The method will always work, if `self._additional_decay_models` is already some kind of iterable it will just chain the two iterables together. This way `self._additional_decay_models` can be called multiple ways or one could sub-class and provide them in the `__init__`. I'll add a test to make sure this works, you're right that this should be tested.

Reply:
Many thanks.